src/services/publisher_service/mod.rs

```diff
@@ -160,15 +185,54 @@ async fn process_queued_messages(
         if let Some(notification) = result {
             let notification_id = notification.subscriber_notification;
             info!("Got a notification with id: {}", notification_id);
-            process_notification(notification, relay_http_client.clone(), metrics, analytics)
-                .await?;
-
-            update_message_processing_status(
-                notification_id,
-                SubscriberNotificationStatus::Published,
-                postgres,
-            )
-            .await?;
+
+            let process_result = process_with_timeout(
+                Duration::from_secs(PUBLISHING_TIMEOUT_MINUTES as u64 * 60),
+                notification,
+                relay_http_client.clone(),
+                metrics,
+                analytics,
+            );
+
+            match process_result.await {
+                Ok(_) => {
+                    update_message_processing_status(
+                        notification_id,
+                        SubscriberNotificationStatus::Published,
+                        postgres,
+                    )
+                    .await?;
+                }
+                Err(e) => match e {
+                    crate::error::Error::TokioTimeElapsed(e) => {
+                        warn!("Timeout elapsed on publishing to the relay: {:?}", e);
+                        update_message_processing_status(
+                            notification_id,
+                            SubscriberNotificationStatus::Queued,
+                            postgres,
+                        )
+                        .await?;
+                    }
+                    crate::error::Error::RelayPublishingError(e) => {
+                        warn!("RelayPublishingError on `process_notification`: {:?}", e);
+                        update_message_processing_status(
+                            notification_id,
+                            SubscriberNotificationStatus::Failed,
+                            postgres,
+                        )
+                        .await?;
+                    }
+                    e => {
```

Comment:
non-blocking thought: we should be able to refactor the errors to eliminate this catch-all block in the future